### PR TITLE
BCDA-2302 Accessibility: Visible focus state

### PR DESCRIPTION
**Pull request by dhgreene**
### Fixes [BCDA-2302](https://jira.cms.gov/browse/BCDA-2302)
The CMS Design System elements until recently had subtle, color-only indications of focus.  The goal of this PR is to take advantage of recent updates to the design system that make focus more obvious.

### Proposed Changes
- Isolate any CSS changes BCDA had made outside of the commented BCDA-specific section
  - Original CSS lines 54-55, 79-80, 7344
- Update CMS Design System Core CSS
  - Original CSS lines 243-4997

Note that the CMS Design System Layout has not been altered (lines 4997-6041).

### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications

This is a format-only change.

#### Examples of the new focus indicators at work
Screen Shot 2020-03-04 at 3 07 57 PM
Screen Shot 2020-03-04 at 3 07 14 PM
Screen Shot 2020-03-04 at 3 06 47 PM
Screen Shot 2020-03-10 at 1 35 38 AM
Screen Shot 2020-03-10 at 1 35 20 AM
Screen Shot 2020-03-10 at 1 35 08 AM

### Feedback Requested
- Please run this locally, as there is a potential impact on unexpected design elements.  Does all look OK?
- Based on the screenshots/local experience, do we need to make any custom changes to visible focus state?  In particular, is the color (new! custom!) for selected links appropriate, or should we choose a different one?

### CODE REVIEW

**Comment by msnook** [comment]
Could you sync this feature branch with `master` so I know I'm pulling the latest when I test locally?

**Comment by dhgreene** [comment]
> Could you sync this feature branch with `master` so I know I'm pulling the latest when I test locally?

@msnook Yes, ready now.

File: assets/css/static-main.css [hunk 1]
```diff
@@ -466,13 +546,89 @@ a:active {
...
+.ds-c-link:visited, a:visited {
+    color: #4c2c92
+}
+
+.ds-c-link:hover, a:hover {
+    color: #205493
+}
+
+.ds-c-link:focus, .ds-c-link:focus:visited, a:focus, a:focus:visited {
+    background-color: #ffde11;
```
Comment (msnook) [inline assets/css/static-main.css]: Is this a color that is required by 508 Compliance, or one that we selected?

Comment (dhgreene) [reply assets/css/static-main.css]: @msnook It's the color chosen by the design system.  I've overridden it in the last commit; do you like this better?

**Comment by dhgreene** [comment]
> (1) Links, once clicked, turn a different color (a color that I would consider to be outside the scope of the visible focus state). For example, in screenshot, I clicked on `home` and now it remains blue/purple. Is that an un-intended side effect?

@msnook I've fixed this for the top menu.  Having the purple color for visited links elsewhere isn't jarring, IMO.

> (2) Links that open in a different window remain in the visible focus state. For example, clicking on the `BCDA Production Onboarding Request Form` opens a new page; but then navigation back to the original page indicates that the focus is still on `BCDA Production Onboarding Request Form`. Is this an un-intended side effect?

This would be intended, because in the original window that link still has focus.  That look/behavior seems more familiar if you try tabbing between links (using a browser that allows tab navigation to links, e.g. NOT Firefox on OS X unless you've changed settings).

**Comment by davidakennedy** [comment]
Hey @dhgreene, thanks for working on this and pushing it forward. Lots of great progress here on what is a big change.

I've retested and have some feedback:

The skip link has some transparency issues when focused. See screenshot.

Screen Shot 2020-03-09 at 3 54 38 PM

The link to the Google Group in the top right corner doesn't meet color contrast requirements when focused. See screenshot.

Screen Shot 2020-03-09 at 3 54 59 PM

The back to top button still has a yellow background. See screenshot.

Screen Shot 2020-03-09 at 3 57 07 PM

The blue background on the horizontal menu doesn't meet minimum color contrast requirements when links are focused. See screenshot.

Screen Shot 2020-03-09 at 4 46 03 PM

The accordions need a visible focus style too. We don't have a component for this in the design system, but the button focus styles should work. But if you all feel that doesn't fit, let me know, and I can help with some ideas. See screenshot.

Screen Shot 2020-03-09 at 4 18 25 PM

Lastly, if I could offer some design feedback on the focus color. If you don't want to go with yellow, to help fit the feel of the site better, maybe try a shade of the green from the gradient in the hero area? Maybe this as the background color: `#68C1BF`? That way, it mimics what the yellow does (it light and bright), but isn't yellow.

That would look something like:

Screen Shot 2020-03-09 at 5 09 47 PM

Let me know if you have any questions!

**Comment by dhgreene** [comment]
@davidakennedy Thanks for catching those details, and for the color suggestion.  The latest commit addresses your review.  Here are some screenshots showing the new behavior:

#### Contrast for links in focus
Screen Shot 2020-03-10 at 1 22 25 AM
#### "Back to top" button styled properly
Screen Shot 2020-03-10 at 1 21 42 AM
#### "Copy to clipboard" button active
Screen Shot 2020-03-10 at 1 21 13 AM
#### Accordion active
Screen Shot 2020-03-10 at 1 20 57 AM
#### Top menu active with proper contrast
Screen Shot 2020-03-10 at 1 20 43 AM
#### Join the Google Group with proper contrast
Screen Shot 2020-03-10 at 1 20 20 AM
#### Skip to main content: opaque
Screen Shot 2020-03-10 at 1 20 09 AM

**Comment by lianilychee** [comment]
lgtm

**Pull request merged**